components/funnel.js

```diff
@@ -0,0 +1,108 @@
+// funnel.js
+
+var _ = require('underscore');
+var fs = require('fs');
+
+var stateFactory = require('../src/create-state.js');
+var logger = require('../src/logger.js');
+var wrapper = require('../src/javascript-wrapper.js');
+
+module.exports = wrapper(funnel);
+
+/**
+ * Funnels a single input into the pipeline one at a time, and queues the rest
+ * Fires the next one when it receives the id that has just finished executing.
```

Comment:
Add comment: WARNING: This will only work when the input is guaranteed to be unique, i.e., never receiving the same ID more than once.  When possible, it should be re-written to better distinguish between the two port connections, perhaps by sniffing the data to distinguish them.